actions/run-nextstrain-build-on-aws-batch/action.yaml

```diff
@@ -0,0 +1,78 @@
+name: Run Nextstrain build on AWS Batch
+description: >-
+  This GitHub Actions action is intended to be called by workflows in our other
+  repos when they need to run a nextstrain build on AWS Batch.
+
+inputs:
+  runtime-envvar-names:
+    description: >-
+      A list of environment variable names for environment variables that
+      should be passed to the build's runtime. The environment variables that
+      the Nextstrain CLI automatically forwards to the runtime do not need to be
```

Comment:
Assuming it's possible to list the environment variables passed to this called workflow via `${{ env }}` or `${{ inputs }}` or somesuch [1], a nicer interface would be for the caller to define the necessary env variables and the called workflow to forward everything to the runtime. The behavior of the nextstrain CLI would then become an implementation detail of this called workflow. For instance,

```yaml
- uses: nextstrain/.github/.../run-builds@v1
  with:
    - AWS_SECRET_ACCESS_KEY=...
    - AWS_ACCESS_KEY_ID=...
    - SOME_WORKFLOW_VAR=...
```

There's discussion above the runtime being ~transparent, and I think this interface makes it easy to imagine how the workflow could run the entire job within a GitHub action (i.e. not using AWS). 

[1] I'm somewhat writing this as if it were a reusable workflow not a composite action, but the idea should be applicable either way.

Reply:
Ah, so pass any/all user defined environment variables to the runtime? 

In the reusable workflow, this would also apply to all `secrets` supplied by the caller. In #44, I'm not pre-defining any secrets in the reusable workflow and asking users to pass all secrets via `secrets: inherit`.

Reply:
[Notes after talking through this with Jover - but the notes are from my perspective!]

I think the caller should have to be explicit about what variables (inputs/secrets) are provided to the build runtime. While GitHub forces us to treat these differently, from the perspective of the build-runtime they're all just env variables. I'd suggest an interface where every build-runtime-needed variable must be defined explicitly by the caller workflow, and the called-workflow (this PR) just passes them through (or prunes out those which the CLI will automagically make available -- that's an implementation detail in my eyes). For instance:

```
- uses: nextstrain/.github/.../run-builds@v1
  with:
    - NCBI_EMAIL=...
  secrets:
    - SLACK_TOKEN=${{ secrets.SLACK_TOKEN }}
    - AWS_SECRET_ACCESS_KEY=${{ secrets.AWS_SECRET_ACCESS_KEY }}
    - AWS_ACCESS_KEY_ID=${{ secrets.AWS_ACCESS_KEY_ID }}
```

While this is more verbose than `secrets: inherit` I think it's much easier to understand given the number of moving parts here.

Reply:
Agreed that `secrets: inherit` can be a bit opaque. This is not something the called workflow can enforce, though, so will be something we do (or don't do) by convention in the calling workflows.

Reply:
OTOH, `secrets: inherit` gives us more central control at the repo and org level without having to make modifications to all the callers.

Reply:
Removed the `runtime-envvar-names` input and passing all explicit envvars and secrets to the build runtime in #44